src/Response/Segment/Segment.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Response\Segment;
+
+use DateTimeImmutable;
+
+class Segment
+{
+    /**
+     * @var non-empty-string
+     */
+    protected string $id;
+
+    /**
+     * @var non-empty-string
+     */
+    protected string $name;
+
+    protected DateTimeImmutable $createdAt;
+
+    protected DateTimeImmutable $updatedAt;
+
+    /**
+     * @var non-empty-string
+     */
+    protected string $appId;
+
+    protected bool $readOnly;
+
+    protected bool $isActive;
+
+    /**
+     * @param non-empty-string $id
+     * @param non-empty-string $name
+     * @param non-empty-string $createdAt
+     * @param non-empty-string $updatedAt
+     * @param non-empty-string $appId
+     */
+    public function __construct(
+        string $id,
+        string $name,
+        string $createdAt,
+        string $updatedAt,
+        string $appId,
+        bool $readOnly,
+        bool $isActive
+    ) {
+        $this->id = $id;
+        $this->name = $name;
+        $this->createdAt = new DateTimeImmutable($createdAt);
+        $this->updatedAt = new DateTimeImmutable($updatedAt);
```

Comment:
I think we should accept DateTimeImmutable as constructor args instead of converting them here